server/core/src/irods_resource_redirect.cpp

```diff
@@ -477,27 +477,14 @@ namespace irods
 
         // =-=-=-=-=-=-=
         // iterate over the list of hostName_t* and see if any match our
-        // host name.  if we do, then were local
-        bool        match_flg = false;
-        hostName_t* tmp_host  = last_resc_host->hostName;
-        while ( tmp_host ) {
-            std::string name( tmp_host->name );
-            if ( name.find( host_name ) != std::string::npos ) {
-                match_flg = true;
-                break;
+        // host name. if we do, then we're local.
+        for (auto* h = last_resc_host->hostName; h; h = h->next) {
+            if (h->name == host_name) {
```

Comment:
certainly not a bad thing, but at this place in the code, should `h->name` ever be a `nullptr`?

i'm thinking no. if the name is a nullptr, then everything breaks and that is fine.

again, this is one of those things i struggle with because i can see it go either way. my gut says we don't need to check it here because we should be free to assume that everything leading up to this point can be trusted.

i can also see doing a check here and logging an error message if we see a nullptr.

what do you feel is the right approach?

Reply:
Sorry, the `h` check was a secondary point there. I was more pointing out the fact that a pointer comparison is not the same as a string comparison (unless I'm just totally missing something here).

To the other points, I feel like I'd feel better if it was checked, but don't feel very strongly about it.

Reply:
I see. `std::string` supports equality comparisons with `char*`'s; same as `std::string_view`.

Reply:
Ooooh, I didn't realize you could use it as the right hand side operator. If that's the case, this is fine, then

Reply:
Yep. That is the reason you see free function versions of `operator==(lhs, rhs)` for classes instead of the member function versions.